environment/src/main/java/com/sequenceiq/environment/environment/service/EnvironmentModificationService.java

```diff
@@ -357,6 +372,34 @@ private void editAdminGroupNameIfChanged(Environment environment, EnvironmentEdi
         }
     }
 
+    private boolean shouldSendSubnetIdsToFreeIpa(AddDnsZoneForSubnetIdsRequest addDnsZoneForSubnetIdsRequest) {
+        AddDnsZoneNetwork addDnsZoneNetwork = addDnsZoneForSubnetIdsRequest.getAddDnsZoneNetwork();
+        return StringUtils.isNotBlank(addDnsZoneNetwork.getNetworkId()) && !addDnsZoneNetwork.getSubnetIds().isEmpty();
```

Comment:
NPE if `addDnsZoneNetwork.getSubnetIds() == null` (which is possible according to `addDnsZoneForSubnetIdsRequest()`, and the field has no explicit initializer).